### DISCUSSION TITLE
github copilot Business - Rate limit

### DISCUSSION THREAD

**Discussion by ymaldor1** [community/community#109329] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Hi,
I've been granted access to github copilot business license last thursday, it's been great.
This morning, it's been working on and off and giving me the rate-limit error :
![image](https://github.com/community/community/assets/126237455/56f66903-0ea3-4b39-a304-c20e108effa0)

From what i understand it's not normal, not sure if someone posted about it already.
First time posting, hopefully did things right. Sorry if I didn't.

Regards,

**Comment by baranelitez** [comment]
I'm on a yearly plan and having this issue as well. First @workspace request of the day and got rate limited.

**Reply by LouisJosso** [reply]
+1

**Reply by stevengonsalvez** [reply]
+1 same error


![image](https://github.com/community/community/assets/9320602/ad42cf34-0ac5-4ab4-bda4-255244ba7f96)

**Comment by hugommoreira** [comment]
+1

**Comment by jamacoe** [comment]
I have Co-Pilot enabled in VS Code and I only use it there. I have a paid account since about half a year and up to today everything worked fine. Since this morning, when asking copilot something in the chat, I receive the message "**Sorry, your request was rate-limited. Please wait and try sending again later.**". The linked information explains that I can do up to 5.000 requests per hour. I think that I haven't had that many request in total since I subscribed. I'm even receiving this message on the very first request after not having requested anything for hours. So what's happening? Has the policy changed? Or has the its enforcement changed lately (and falsely)?

**Comment by MJWexer** [comment]
+1 from me.
rate limited after 25 (1 exceedingly big i.e many tokens) requests. I accidentally CPed something big into the chat.

**Reply by MJWexer** [reply]
I waited half an hour and then I was able to query again

**Comment by rockyshi1993** [comment]
I'm experiencing exactly the same issue.

I’m on a paid Copilot plan and mostly using it in VS Code, with relatively light usage. However, I still get rate limited even on the very first request after being idle for hours.

From what I understand, this doesn’t seem to be related to the documented request limits (e.g. thousands of requests per hour), but more likely due to internal rate limiting mechanisms such as short-term burst limits, backend capacity, or model-specific restrictions.

According to the docs, rate limits can also be triggered by system load or fairness policies, not just total request count. :contentReference[oaicite:0]{index=0}

It would really help if GitHub could clarify:
- what exact conditions trigger these limits
- whether there are hidden per-minute / per-session limits
- and why users can get rate limited even with minimal usage

Right now it feels unpredictable and makes Copilot unreliable for real work.

Would appreciate any official clarification on this.
